### ISSUE TITLE
Open new tabs instead of new windows

### ISSUE THREAD

**Issue by parisni** [luakit/luakit#509]
Hey

luakit provides tabs but I prefer to use an other windows manager.  Then I press "w" instead of "t" to open in a new windows.

Unfortunately, bookmarks does not provide a shortkey to be opened in a new window. Maybe there is many other features that do not provide those mecanisms too.

For now, I use "w" luakit://bookmarks to open it in a new window. 

Does this new feature make sense ?

Thanks

**Comment by jonas-schievink** [comment]
You can just rebind any key to do what you want, that's the point of this browser. What additional feature do you need?

**Comment by parisni** [comment]
that's right I will rebind. However I guess all action could be done in 3 way natively :
- same window
- new tab
- new window

**Comment by aidanholm** [comment]
Hey @parisni!
Perhaps you'd like a way to tell luakit to always open things in a new window, instead of in a new tab? So `:winopen` and `:tabopen` would both just open a new window?

**Comment by parisni** [comment]
hey @aidanholm 

that sounds good. Firefox does have such feature "open in all in a new windows" .
That would make sense with i3 I guess.

**Comment by jonas-schievink** [comment]
@aidanholm Title of this issue is backwards

**Comment by rti** [comment]
Hi,

I did the following to open everything in a new window instead of a tab:

```
luakit: open all links in a new window

diff --git a/.config/luakit/window.lua b/.config/luakit/window.lua
index d10d6a2..7baef52 100644
--- a/.config/luakit/window.lua
+++ b/.config/luakit/window.lua
@@ -394,6 +394,12 @@ _M.methods = {
     end,
 
     new_tab = function (w, arg, opts)
+        local window = require("window")
+        local win = window.new({arg})
+        return win.view
+    end,
+
+    new_tab_orig = function (w, arg, opts)
         opts = opts or {}
         assert(type(opts) == "table")
         local switch, order = opts.switch, opts.order
@@ -446,7 +452,7 @@ _M.methods = {
         end
         -- Treat a blank last tab as an empty notebook (if blank_last=true)
         if blank_last ~= false and w.tabs:count() == 0 then
-            w:new_tab(settings.window.new_tab_page, false)
+            w:new_tab_orig(settings.window.new_tab_page, false)
         end
     end,
 
@@ -506,7 +512,7 @@ _M.methods = {
     -- Navigate current view or open new tab
     navigate = function (w, arg, view)
         view = view or w.view
-        if not view then return w:new_tab(arg) end
+        if not view then return w:new_tab_orig(arg) end
         require("webview").set_location(view, arg)
     end,
 
@@ -679,7 +685,7 @@ function _M.new(args)
         if type(arg) == "string" then
             arg = w:search_open(arg)
         end
-        w:new_tab(arg, false)
+        w:new_tab_orig(arg, false)
     end
 
     -- Show window
@@ -690,7 +696,7 @@ function _M.new(args)
 
     -- Make sure something is loaded
     if w.tabs:count() == 0 then
-        w:new_tab(w:search_open(settings.window.home_page), false)
+        w:new_tab_orig(w:search_open(settings.window.home_page), false)
     end
 
     return w

```

My changes are based on `window.lua` from `/usr/share/luakit/lib/window.lua` as installed by arch linux's aur package `luakit-git 2017.08.10.r181.g3ff1e33f-1`.

Disclaimer: This is my very first attempt to change luakit's stock behaviour. I might be doing it completely wrong.

**Comment by aidanholm** [comment]
That's a perfectly acceptable way to mod luakit, as long as you're happy to manually update your patched file.

Your changes are a bit different to the change I had in mind, which is to check if there's already a tab in `w:new_tab()`, and if so to make a new window instead of a new tab.

A fully-fledged solution would have a setting, would hide the tab bar, etc. etc. There are probably going to be messy corner cases with tab detaching and private tabs as well... e.g. it's currently not possible to open a window with its tab starting as private

**Comment by arcnmx** [comment]
[FWIW here is my own approach](https://github.com/arcnmx/dotfiles/blob/2a977f94616e57535eddf141072cc81e6a1b61f5/files/gui/.config/luakit/rc/ui.lua#L7) that doesn't modify luakit files directly.

**Comment by aidanholm** [comment]
That's closer to the modding style I'd generally use, as it doesn't require manual merges of new changes.

I will probably eventually get around to working on including this in luakit properly, with a `window.tabs_are_windows` setting.

**Comment by c0dev0id** [comment]
The issue is 12 months old and the solution has been implemented. I'm closing this now.

**Comment by arcnmx** [comment]
Could you please reference the commit or setting or something about where this was fixed/implemented? I'm just seeing a bunch of issues being closed because incomplete/hacky workarounds were provided in the comments, which is kind of confusing?

**Comment by c0dev0id** [comment]
We have many issues that are from previous generations of luakit as well as many that are solved by providing configuration, but authors are never reacting. I'm working on getting the issue list back in check.

I might have gotten confused with this one, as the referenced issue got solve.

Reopening this one (this is always an option when it is still relevant).

**Comment by ghost** [comment]
Are there any plans to implement a tabs_are_windows feature? I would love it if this gets implemented.
